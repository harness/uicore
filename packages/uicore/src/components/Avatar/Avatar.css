/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.Avatar {
  display: inline-block;
  margin: 0 5px;
}

.AvatarInner {
  font-weight: bolder;
  box-sizing: border-box;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: flex;
  align-items: center;
  justify-content: center;
}
.contentStyle {
  text-align: center;
  cursor: pointer;
}
.imageStyle {
  display: block;
}
.hoverToolTipLayout {
  word-break: break-all !important;
  text-align: start !important;
  max-width: 320px;
  min-width: 206px;
}
.hoverAvatarLayout {
  justify-content: flex-start !important;
  column-gap: 5px;
  background: var(--grey-100) !important;
  height: 54px;
  padding-bottom: 0px !important;
  align-items: flex-end !important;
}
.avatarPopOver {
  display: flex;
  align-items: center;
}
.avatarImg {
  position: relative;
  top: 11px;
}
