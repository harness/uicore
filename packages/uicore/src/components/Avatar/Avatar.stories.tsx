/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { Layout, Avatar, AvatarProps } from '../..'

export default {
  title: 'Components / Avatar',
  component: Avatar
} as Meta

export const Normal: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar
      name="John Doe"
      email="John.Doe@abc.com"
      onClick={() => {
        // eslint-disable-next-line
        alert(' Avatar clicked')
      }}
      size={'normal'}
      hoverCardDetailsCallBack={() => {
        // eslint-disable-next-line
        alert(' Details clicked')
      }}
      {...args}
    />
  </Layout.Horizontal>
)

export const Small: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar email="ajohn.doe@harnes.io" name="aJoh Doe" size={'small'} {...args} />
  </Layout.Horizontal>
)

export const Xsmall: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar email="bjohn.doe@harnes.io" name="bJoh Doe" size={'xsmall'} {...args} />
  </Layout.Horizontal>
)

export const Medium: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar email="cjohn.doe@harnes.io" name="cJoh Doe" size={'medium'} {...args} />
  </Layout.Horizontal>
)

export const Large: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar email="djohn.doe@harnes.io" name="dJoh Doe" size={'large'} {...args} />
  </Layout.Horizontal>
)

export const BorderRadius: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar email="fjohn.doe@harnes.io" borderRadius={25} {...args} />
  </Layout.Horizontal>
)

export const Image: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar
      src="https://image.shutterstock.com/image-photo/buddha-statue-hyderabadhyderabadindia-260nw-1366155431.jpg"
      size={'large'}
      hoverCard={false}
      {...args}
    />
  </Layout.Horizontal>
)

// Edge case for testing, this should render user icon without any initials
export const NullName: Story<AvatarProps> = args => (
  <Layout.Horizontal spacing="small" id="primary-buttons">
    <Avatar name={null as any} {...args} />
  </Layout.Horizontal>
)
