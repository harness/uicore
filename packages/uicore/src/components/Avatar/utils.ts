/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { pick, defaultTo } from 'lodash-es'
import { Color } from '@harness/design-system'

export const getInitialsFromNameOrEmail = (name = '', email = ''): string => {
  let initialsFromName = defaultTo(
    name
      ?.split(/-| /)
      .map((n: string) => n[0])
      .join(''),
    ''
  )
  initialsFromName =
    initialsFromName.length > 2
      ? `${initialsFromName[0]}${initialsFromName[initialsFromName.length - 1]}`
      : initialsFromName

  let initialsFromEmail = defaultTo(email?.split('@')[0], '')
  const splitedInitialsFromEmail = initialsFromEmail.split('.')
  initialsFromEmail =
    splitedInitialsFromEmail.length > 1
      ? `${splitedInitialsFromEmail[0][0]}${splitedInitialsFromEmail[splitedInitialsFromEmail.length - 1][0]}`
      : initialsFromEmail[0]

  return initialsFromName || initialsFromEmail
}

export const getSumOfAllCharacters = (str: string): number => {
  return str
    ? str
        .toLowerCase()
        .split('')
        .reduce((initialValue, charc: string) => {
          return initialValue + (charc.charCodeAt(0) - 'a'.charCodeAt(0)) + 1
        }, 0)
    : 0
}

export const defaultAvatarColor = Object.values(
  pick(Color, [
    'GREY_600',
    'GREY_700',
    'GREY_850',
    'GREY_900',
    'BLUE_700',
    'BLUE_800',
    'RED_700',
    'RED_600',
    'YELLOW_800',
    'YELLOW_700',
    'YELLOW_600',
    'PURPLE_900',
    'ORANGE_500',
    'GREEN_500',
    'GREEN_600',
    'GREEN_700'
  ])
)
