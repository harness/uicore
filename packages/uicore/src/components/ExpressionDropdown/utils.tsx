/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { IconName } from '@harness/icons'
import { isEmpty } from 'lodash-es'
import { TrieNode } from './ExpressionDropdown'

export function setOpenTargetElement(
  isOpen: boolean[],
  setIsOpen: React.Dispatch<React.SetStateAction<boolean[]>>,
  valueTillHere: string,
  targetValue: boolean
): void {
  const newIsOpen = [...isOpen]
  let targetElement = valueTillHere.split('.').length
  if (isEmpty(valueTillHere)) {
    targetElement = targetElement - 1
  }
  if (newIsOpen[targetElement] === targetValue) {
    newIsOpen[targetElement] = !targetValue
  } else {
    newIsOpen[targetElement] = targetValue
  }
  setIsOpen(newIsOpen)
}

export function getDropDownIcon(item: TrieNode, isOpen: boolean[]): IconName {
  let targetElement
  if (isEmpty(item.valueTillHere)) {
    targetElement = item.valueTillHere.split('.').length - 1
  } else {
    targetElement = item.valueTillHere.split('.').length
  }
  if (isOpen[targetElement]) {
    return 'main-chevron-down'
  }
  return 'main-chevron-right'
}
