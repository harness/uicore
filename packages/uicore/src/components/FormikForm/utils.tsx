/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { ReactNode, useContext } from 'react'
import { FormikContextType } from 'formik'

import { get } from 'lodash-es'
import { FormikTooltipContext } from './FormikTooltipContext'
import { HarnessDocTooltip } from '../../frameworks/Tooltip/Tooltip'
import { DataTooltipInterface } from '../../frameworks/Tooltip/types'

export interface FormikExtended<T> extends FormikContextType<T> {
  disabled?: boolean
  inline?: boolean
  formName: string
}

export interface FormikContextProps<T> {
  formik?: FormikExtended<T>
  optionalLabel?: string
  isOptional?: boolean // default to false
  autoComplete?: string
  tooltipProps?: DataTooltipInterface // todo mark it as mandatory
}

export const IsOptionLabel = '(optional)'

const isObject = (obj: any): boolean => obj !== null && typeof obj === 'object'

export const errorCheck = (name: string, formik?: FormikContextType<any>): boolean =>
  Boolean(
    (get(formik?.touched, name) || (formik?.submitCount && formik?.submitCount > 0)) &&
      get(formik?.errors, name) &&
      !isObject(get(formik?.errors, name))
  )

export const getFormFieldLabel = (
  label: ReactNode | string | undefined,
  fieldName: string,
  props: FormikContextProps<any>,
  css?: string
): ReactNode | string | undefined => {
  const optionalLabel = props.optionalLabel || IsOptionLabel
  const labelText = !props.isOptional ? label : `${label} ${optionalLabel}`
  if (!labelText) {
    return labelText
  }
  const tooltipContext = useContext(FormikTooltipContext)
  const dataTooltipId =
    props.tooltipProps?.dataTooltipId || (tooltipContext?.formName ? `${tooltipContext?.formName}_${fieldName}` : '')
  return <HarnessDocTooltip tooltipId={dataTooltipId} labelText={labelText} className={css || ''} />
}

export function escapeNewlines(input: string | number): string | number {
  if (typeof input === 'string') {
    return input.replace(/\n/g, '\\n').replace(/\r/g, '\\r')
  }
  return input
}
