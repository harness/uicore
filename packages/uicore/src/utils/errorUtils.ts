/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { isEmpty } from 'lodash-es'

const IGNORED_ERRORS = [
  'Failed to fetch: The user aborted a request.',
  'Failed to fetch: Failed to fetch',
  'Failed to fetch: 504 Gateway Timeout',
  "Failed to fetch: Failed to execute 'fetch' on 'Window': The user aborted a request.",
  'Failed to fetch: The operation was aborted.',
  'Failed to fetch: Fetch is aborted',
  'Failed to fetch: signal is aborted without reason'
]

export function shouldShowError(e: any): boolean {
  const hideMessagesForStatusCodes = [502, 503]
  const message = (e?.message || '').trim()

  return !IGNORED_ERRORS.includes(message) && !hideMessagesForStatusCodes.includes(e?.status)
}

/* TODO Don't see proper types for this new errors format, replace Record<string, any> with more stricter type when available */
export function getErrorInfoFromErrorObject(error: Record<string, any>, errorArrayAsPriority?: boolean): string {
  let errorArrayValStr = ''
  if (!isEmpty(error?.data?.errors) && Array.isArray(error?.data?.errors)) {
    const arrVal = error.data.errors as Array<{ fieldId: string; error: string }>
    errorArrayValStr = arrVal
      .map(err => {
        return `${err.fieldId} ${err.error}`
      })
      .join(', ')
  }
  if (errorArrayAsPriority && errorArrayValStr) {
    return errorArrayValStr
  }
  return error?.data?.message || errorArrayValStr || error?.message || ''
}
