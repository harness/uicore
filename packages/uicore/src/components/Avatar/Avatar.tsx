/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { HTMLDivProps, Popover } from '@blueprintjs/core'
import React, { useState } from 'react'
import css from './Avatar.css'
import classnames from 'classnames'
import { getInitialsFromNameOrEmail, getSumOfAllCharacters, defaultAvatarColor } from './utils'
import { FontSize } from '@harness/design-system'
import { Utils } from '../../core/Utils'
import { Color } from '@harness/design-system'
import { Container } from '../../components/Container/Container'
import { Layout } from '../../layouts/Layout'
import { Text } from '../Text/Text'
import { Icon, IconProps } from '@harness/icons'
export type AvatarSizes = FontSize
export interface AvatarProps extends HTMLDivProps {
  name?: string
  src?: string
  backgroundColor?: Color | Color[]
  email?: string
  size?: AvatarSizes
  borderRadius?: number
  borderColor?: Color
  color?: Color
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void
  hoverCard?: boolean
  tooltip?: JSX.Element
  usePortal?: boolean
  hoverCardDetailsCallBack?: (e: React.MouseEvent<HTMLDivElement>) => void
}
export const plusIcon = '+'
interface SizeValueProps {
  size: string
  fontSize: string
  lineHeight: string
  imageHeight: string
  plusIconLineHeight: string
}
type SizesProps = {
  [key in AvatarSizes]: SizeValueProps
}
const sizes: SizesProps = {
  xsmall: {
    size: '16px',
    fontSize: '7px',
    lineHeight: '12px',
    imageHeight: '12px',
    plusIconLineHeight: '10px'
  },
  small: {
    size: '24px',
    fontSize: '10px',
    lineHeight: '18px',
    imageHeight: '20px',
    plusIconLineHeight: '16px'
  },
  normal: {
    size: '32px',
    fontSize: '10px',
    lineHeight: '26px',
    imageHeight: '28px',
    plusIconLineHeight: '25px'
  },
  medium: {
    size: '48px',
    fontSize: '18px',
    lineHeight: '42px',
    imageHeight: '44px',
    plusIconLineHeight: '40px'
  },
  large: {
    size: '72px',
    fontSize: '28px',
    lineHeight: '66px',
    imageHeight: '68px',
    plusIconLineHeight: '60px'
  }
}

export const Avatar: React.FC<AvatarProps> = (props: AvatarProps) => {
  const [loadFailed, setLoadFailed] = useState(false)
  const onLoadFailed = () => setLoadFailed(true)
  const {
    borderRadius = 100,
    src,
    name = '',
    backgroundColor = defaultAvatarColor,
    size = 'normal',
    style,
    className,
    email = '',
    tooltip,
    usePortal = false,
    color = Color.WHITE,
    onClick,
    hoverCard = true,
    borderColor = Color.WHITE,
    hoverCardDetailsCallBack,
    ...rest
  } = props

  let textColor = color
  const formatedSize = sizes[size].size
  const fontSize = sizes[size].fontSize
  const lineHeight = sizes[size].lineHeight
  const imageHeightWidth = sizes[size].imageHeight
  let forcePreventHover = false
  let inner
  let initials = ''
  if (!src) {
    initials = getInitialsFromNameOrEmail(name, email)
  }
  // sum of characters will be used to decide the background color for that avatar
  const sumOfCharacters = Math.abs(getSumOfAllCharacters(initials))
  let calucatedBackgroundColor = Array.isArray(backgroundColor)
    ? backgroundColor[sumOfCharacters % backgroundColor.length]
    : backgroundColor
  if (textColor === calucatedBackgroundColor) {
    textColor = Color.WHITE
    calucatedBackgroundColor = Color.BLACK
  }
  const contentStyle = {
    borderRadius: `${borderRadius}%`,
    lineHeight,
    width: formatedSize,
    height: formatedSize,
    ...(!src && {
      backgroundColor: Utils.getRealCSSColor(calucatedBackgroundColor || 'BLUE_800'),
      color: Utils.getRealCSSColor(textColor)
    }),
    border: `2px solid ${Utils.getRealCSSColor(borderColor)}`,
    ...(fontSize && { fontSize })
  }
  if (src) {
    const parsedHeightWidth = parseInt(imageHeightWidth, 10) as IconProps['size']
    const imageStyleNew = { ...contentStyle, width: imageHeightWidth, height: imageHeightWidth, border: '' }
    inner = loadFailed ? (
      <Icon name="user" size={parsedHeightWidth} />
    ) : (
      <img src={src} style={imageStyleNew} className={css.imageStyle} alt={name} onError={onLoadFailed} />
    )
  } else {
    if (!initials) {
      // show empty user Icon if neither 'src', nor 'initials' are truthy
      const iconSize = parseInt(sizes[size].fontSize, 10) as IconProps['size']
      inner = <Icon name="user" color={Color.WHITE} size={iconSize} />
      forcePreventHover = true
    } else {
      inner = initials.toUpperCase()
    }
  }
  const toolTipStyle = {
    ...contentStyle,
    width: sizes.medium.size,
    height: sizes.medium.size,
    fontSize: sizes.medium.fontSize,
    lineHeight: sizes.medium.lineHeight,
    borderRadius: '100%'
  }

  const parsedHeightWidthInTooltip = parseInt(sizes.medium.size, 10) as IconProps['size']

  const defaultTooltip = (
    <Layout.Vertical className={css.hoverToolTipLayout}>
      <Layout.Horizontal flex padding="medium" className={css.hoverAvatarLayout}>
        {src ? (
          loadFailed ? (
            <Icon name="user" size={parsedHeightWidthInTooltip} className={css.avatarImg} />
          ) : (
            <img
              className={css.avatarImg}
              src={src}
              style={{ ...toolTipStyle, textAlign: 'center', minWidth: sizes.medium.size }}
              alt={name}
              onError={onLoadFailed}
            />
          )
        ) : (
          <div className={css.avatarImg} style={{ ...toolTipStyle, textAlign: 'center', minWidth: sizes.medium.size }}>
            {inner}
          </div>
        )}

        <Text margin={{ bottom: 'small' }} font={{ size: 'small', weight: 'bold' }} color={Color.BLACK} lineClamp={1}>
          {name}
        </Text>
      </Layout.Horizontal>

      {(email || hoverCardDetailsCallBack) && (
        <Container padding="medium">
          {email && (
            <Text
              margin={{ top: 'small' }}
              font={{ size: 'small' }}
              icon="email-inline"
              iconProps={{ color: 'grey600', size: 10 }}
              lineClamp={1}
              className={css.emailHover}>
              {email}
            </Text>
          )}
          {hoverCardDetailsCallBack && (
            <Text
              font={{ size: 'xsmall' }}
              margin={{ top: 'small' }}
              color="primary7"
              lineClamp={1}
              onClick={hoverCardDetailsCallBack}>
              Details
            </Text>
          )}
        </Container>
      )}
    </Layout.Vertical>
  )
  let hoverCardContent = {}
  if (hoverCard) {
    hoverCardContent = { content: tooltip || defaultTooltip }
  }
  const getUpdatedContentStyle = () => {
    if (name === plusIcon) {
      return {
        ...contentStyle,
        fontSize: `${2 * parseInt(sizes[size].fontSize, 10)}px`,
        lineHeight: sizes[size].plusIconLineHeight,
        fontWeight: 200
      }
    } else {
      return contentStyle
    }
  }
  const isHoverDisabled = !hoverCard || forcePreventHover
  return (
    <div className={classnames(className, css.Avatar, css.contentStyle)} style={style} onClick={onClick} {...rest}>
      <Popover
        {...hoverCardContent}
        interactionKind="hover"
        usePortal={usePortal}
        className={css.avatarPopOver}
        position="top"
        disabled={isHoverDisabled}>
        <div className={css.AvatarInner} style={getUpdatedContentStyle()}>
          {inner}
        </div>
      </Popover>
    </div>
  )
}
