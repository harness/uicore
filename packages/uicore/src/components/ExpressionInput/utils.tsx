/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ChildKeyNode } from 'components/ExpressionDropdown/ExpressionDropdown'

interface TrieNode {
  value: string
  valueTillHere: string
  children: TrieNode[]
  childKeys: ChildKeyNode[]
  childExpressions: string[]
}

export function formatData(expressions: string[]): TrieNode {
  const rootTrieNode: TrieNode = {
    value: 'Main',
    valueTillHere: '',
    children: [],
    childKeys: [],
    childExpressions: []
  }
  expressions.forEach(expression => {
    const expressionWords = expression.split('.')
    let startNode = rootTrieNode

    expressionWords.forEach((word, index) => {
      startNode.childExpressions.push(expression)
      const isWordPresent = startNode.children.findIndex(child => child.value === word)
      if (isWordPresent !== -1) {
        startNode = startNode.children[isWordPresent]
      } else {
        let valueTillHere = ''
        expressionWords.forEach((w, ind) => {
          if (ind <= index) {
            if (valueTillHere.length) {
              valueTillHere = valueTillHere + '.' + w
            } else {
              valueTillHere = w
            }
          }
        })
        startNode.childKeys.push({
          key: word,
          value: valueTillHere
        })
        startNode.children.push({
          value: word,
          valueTillHere: valueTillHere,
          children: [],
          childKeys: [],
          childExpressions: []
        })
        startNode = startNode.children[startNode.children.length - 1]
      }
    })
  })
  return rootTrieNode
}
