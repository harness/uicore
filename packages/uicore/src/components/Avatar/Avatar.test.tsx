/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { Avatar } from './Avatar'
import { getInitialsFromNameOrEmail, getSumOfAllCharacters } from './utils'

import { Color } from '@harness/design-system'

describe('Render basic component', () => {
  test('should check snapshot with name', () => {
    const { container } = render(<Avatar backgroundColor={Color.RED_100} name="John Doe" />)
    expect(container).toMatchSnapshot()
  })
  test('should check snapshot with email', () => {
    const { container } = render(<Avatar backgroundColor={Color.RED_100} email="John.Doe@harness.io" />)
    expect(container).toMatchSnapshot()
  })
  test('should check snapshot with image', () => {
    const { container } = render(
      <Avatar backgroundColor={Color.RED_100} src="https://www.mangaship.com/Content/img/human-avatar.png" />
    )
    expect(container).toMatchSnapshot()
  })
  test('getInitials with name', () => {
    expect(getInitialsFromNameOrEmail('All Name Here')).toEqual('AH')
    expect(getInitialsFromNameOrEmail('All Name')).toEqual('AN')
    expect(getInitialsFromNameOrEmail('AllNameHere')).toEqual('A')
    // eslint-disable-next-line
    // @ts-ignore
    expect(getInitialsFromNameOrEmail('AllNameHere', null)).toEqual('A')
  })
  test('getInitials with email', () => {
    expect(getInitialsFromNameOrEmail('', 'AllNameHere@com')).toEqual('A')
    expect(getInitialsFromNameOrEmail('', 'AllName.here@com')).toEqual('Ah')
    expect(getInitialsFromNameOrEmail(undefined, 'abc')).toEqual('a')
    // eslint-disable-next-line
    // @ts-ignore
    expect(getInitialsFromNameOrEmail(null, 'abc')).toEqual('a')
    // eslint-disable-next-line
    // @ts-ignore
    expect(getInitialsFromNameOrEmail(null, null)).toEqual(undefined)
  })
  test('getSumofCharacters', () => {
    expect(getSumOfAllCharacters('abcd')).toEqual(10)
    expect(getSumOfAllCharacters('Abcd')).toEqual(10)
    expect(getSumOfAllCharacters('')).toEqual(0)
  })
})
