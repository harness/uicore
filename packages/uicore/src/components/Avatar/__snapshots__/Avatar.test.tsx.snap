// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render basic component should check snapshot with email 1`] = `
<div>
  <div
    class="Avatar contentStyle"
  >
    <span
      class="bp3-popover-wrapper avatarPopOver"
    >
      <span
        class="bp3-popover-target"
      >
        <div
          class="AvatarInner"
          style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
          tabindex="0"
        >
          JD
        </div>
      </span>
    </span>
  </div>
</div>
`;

exports[`Render basic component should check snapshot with image 1`] = `
<div>
  <div
    class="Avatar contentStyle"
  >
    <span
      class="bp3-popover-wrapper avatarPopOver"
    >
      <span
        class="bp3-popover-target"
      >
        <div
          class="AvatarInner"
          style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
          tabindex="0"
        >
          <img
            alt=""
            class="imageStyle"
            src="https://www.mangaship.com/Content/img/human-avatar.png"
            style="border-radius: 100%; line-height: 26px; width: 28px; height: 28px; font-size: 10px;"
          />
        </div>
      </span>
    </span>
  </div>
</div>
`;

exports[`Render basic component should check snapshot with name 1`] = `
<div>
  <div
    class="Avatar contentStyle"
  >
    <span
      class="bp3-popover-wrapper avatarPopOver"
    >
      <span
        class="bp3-popover-target"
      >
        <div
          class="AvatarInner"
          style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
          tabindex="0"
        >
          JD
        </div>
      </span>
    </span>
  </div>
</div>
`;
